network/consensus/base.go

```diff
@@ -24,28 +24,66 @@ import (
 	"github.com/insolar/insolar/log"
 )
 
-type exchangeResults map[core.RecordRef][]*core.ActiveNode
+// exchangeResults is thread safe results struct
+type exchangeResults struct {
+	mutex *sync.RWMutex
+	data  map[core.RecordRef][]*core.ActiveNode
+	hash  []*NodeUnsyncHash
+}
+
+func (r *exchangeResults) writeResultData(id core.RecordRef, data []*core.ActiveNode) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	r.data[id] = data
+}
+
+func (r *exchangeResults) getData(id core.RecordRef) []*core.ActiveNode {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	return r.data[id]
+}
+
+func (r *exchangeResults) calculateResultHash() []*NodeUnsyncHash {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	for id, x := range r.data {
+		d, err := CalculateNodeUnsyncHash(id, x)
+		if err != nil {
+			log.Error(err)
+			continue
```

Comment:
Is it ok to skip wrong calculation and continue working? May be method calculateResultHash should return error in that case?

Reply:
most likely there will never be an error. in consensus algorithm it's not very important if hash does not calculated

Reply:
I mean, it looks like  r.data and r.hash should be the same length. If it is, and and one hash is not calculated - result may be surprising =)